### PR TITLE
Substitute props.intl.formatMessage with <FormattedMessage />

### PR DESCRIPTION
**Pull request by michizhou**
Replaced several instances of `props.intl.formatMessage` with `<FormattedMessage />` where a string is passed for the pages rendered by `UserInfoEdit.js` and `SlackConfig.js`
Helps to resolve meedan/check#20

### CODE REVIEW

File: src/app/components/source/UserInfoEdit.js [hunk 1]
```diff
@@ -482,7 +485,10 @@ class UserInfoEdit extends React.Component {
                   onClick={this.handleEditProfileImg.bind(this)}
                   color="primary"
                 >
-                  {this.props.intl.formatMessage(globalStrings.edit)}
+                  <FormattedMessage 
+                    id={globalStrings.edit.id} 
+                    defaultMessage={globalStrings.edit.defaultMessage} 
+                  />
```
Comment (amoedoamorim) [inline src/app/components/source/UserInfoEdit.js:491]: Here you should use `<FormattedMessage {...globalStrings.edit} />`

File: src/app/components/source/UserInfoEdit.js [hunk 2]
```diff
@@ -553,7 +559,10 @@ class UserInfoEdit extends React.Component {
                   color="primary"
                   onClick={this.handleAddLink.bind(this)}
                 >
-                  {this.props.intl.formatMessage(messages.addLink)}
+                  <FormattedMessage 
+                    id={messages.addLink.id} 
+                    defaultMessage={messages.addLink.defaultMessage}
```
Comment (amoedoamorim) [inline src/app/components/source/UserInfoEdit.js:564]: The desired usage is that the actual content of these keys is moved here, and `addLink` is to be removed from the `messages` object.

File: src/app/components/source/UserInfoEdit.js [hunk 3]
```diff
@@ -562,15 +571,21 @@ class UserInfoEdit extends React.Component {
                   className="source__edit-cancel-button"
                   onClick={handleLeaveEditMode}
                 >
-                  {this.props.intl.formatMessage(globalStrings.cancel)}
+                  <FormattedMessage 
+                    id={globalStrings.cancel.id} 
+                    defaultMessage={globalStrings.cancel.defaultMessage} 
+                  />
```
Comment (amoedoamorim) [inline src/app/components/source/UserInfoEdit.js:577]: Here you should use `<FormattedMessage {...globalStrings.cancel} />`

File: src/app/components/source/UserInfoEdit.js [hunk 4]
```diff
@@ -562,15 +571,21 @@ class UserInfoEdit extends React.Component {
...
                   variant="contained"
                   className="source__edit-save-button"
                   color="primary"
                   onClick={this.handleSubmit.bind(this)}
                 >
-                  {this.props.intl.formatMessage(globalStrings.save)}
+                  <FormattedMessage 
+                    id={globalStrings.save.id} 
+                    defaultMessage={globalStrings.save.defaultMessage} 
+                  />
```
Comment (amoedoamorim) [inline src/app/components/source/UserInfoEdit.js:588]: Here you should use `<FormattedMessage {...globalStrings.save} />`

File: src/app/components/team/SlackConfig.js [hunk 5]
```diff
@@ -166,7 +166,10 @@ class SlackConfig extends React.Component {
           onClose={this.handleCloseDialog}
         >
           <DialogTitle>
-            {this.props.intl.formatMessage(messages.title)}
+            <FormattedMessage 
+              id={messages.title.id} 
+              defaultMessage={messages.title.defaultMessage}
```
Comment (amoedoamorim) [inline src/app/components/team/SlackConfig.js:171]: The desired usage is that the actual content of these keys is moved here, and `title` is to be removed from the `messages` object.

File: src/app/components/team/SlackConfig.js [hunk 6]
```diff
@@ -195,10 +198,16 @@ class SlackConfig extends React.Component {
           </DialogContent>
           <DialogActions>
             <Button onClick={this.handleCloseDialog}>
-              {this.props.intl.formatMessage(globalStrings.cancel)}
+              <FormattedMessage 
+                id={globalStrings.cancel.id} 
+                defaultMessage={globalStrings.cancel.defaultMessage} 
+              />
```
Comment (amoedoamorim) [inline src/app/components/team/SlackConfig.js:204]: Here you should use `<FormattedMessage {...globalStrings.cancel} />`

File: src/app/components/team/SlackConfig.js [hunk 7]
```diff
@@ -195,10 +198,16 @@ class SlackConfig extends React.Component {
...
             </Button>
             <Button color="primary" onClick={this.handleSubmit.bind(this)}>
-              {this.props.intl.formatMessage(globalStrings.save)}
+              <FormattedMessage 
+                id={globalStrings.save.id} 
+                defaultMessage={globalStrings.save.defaultMessage} 
+              />
```
Comment (amoedoamorim) [inline src/app/components/team/SlackConfig.js:210]: Here you should use `<FormattedMessage {...globalStrings.save} />`

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Hello @michizhou ! Thanks so much for your interest in contributing, we appreciate it! In order for us to be able to accept your PR please apply the requested changes! Have a great Hacktoberfest!

**Comment by michizhou** [comment]
@amoedoamorim I have made some of the requested changes, though it seems the SlackConfig.js file has been deleted and reorganized into an index.js file that handles some of the changes already (I already resolved the merge conflict). Please let me know if any additional changes or modifications need to be made. Thank you.

File: src/app/components/source/UserInfoEdit.js [hunk 8]
```diff
@@ -447,7 +438,10 @@ class UserInfoEdit extends React.Component {
             {link.error ?
               null :
               <StyledHelper>
-                {this.props.intl.formatMessage(messages.addLinkHelper)}
+                <FormattedMessage
```
Comment (amoedoamorim) [inline src/app/components/source/UserInfoEdit.js:441]: Please remove trailing spaces

File: src/app/components/source/UserInfoEdit.js [hunk 9]
```diff
@@ -553,7 +547,10 @@ class UserInfoEdit extends React.Component {
                   color="primary"
                   onClick={this.handleAddLink.bind(this)}
                 >
-                  {this.props.intl.formatMessage(messages.addLink)}
+                  <FormattedMessage
```
Comment (amoedoamorim) [inline src/app/components/source/UserInfoEdit.js:550]: Please remove trailing spaces

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Thanks @michizhou . There are still some minor details that need fixing, which are some left trailing spaces, and there's a major one: `FormattedMessage` is not defined. Please include it in the import statement in line 4.

**Comment by michizhou** [comment]
@amoedoamorim My apologies for not catching those minor errors in my prior commits. I have made the necessary changes in my latest commit above. If there is anything else that needs to be fixed, please let me know.

**Review by amoedoamorim — APPROVED** [review]
Thanks for contributing @michizhou ! We appreciate it! Your PR looks good and will be merged!

**Comment by danielafeitosa** [comment]
Hi @michizhou!

For your contribution to CheckWeb during [Meedan Hacktoberfest 2020](https://meedan.com/hacktoberfest) you are eligible to earn a swag :tada: 
To receive it please fill in the form with your information: https://forms.gle/qKMRkWK7CEccCtGv5

Thank you!

**Pull request merged**